### PR TITLE
synapser refactor phase a

### LINKED ISSUE #94: Refactor synapse-adjacent functions to use new reticulate-based synapser
We should drop the custom import of the synapse python client in lieu of the new reticulate-based synapser, which likely will have better ongoing support (at least compared to us maintaining a custom import of the client) and will make it easier for users of this package to import in additional synapser functions/R docs instead of having to use synapseclient via reticulate.

### LINKED ISSUE #173: Phase A for synapser factor
Factor out Phase A of https://github.com/nf-osi/nfportalutils/issues/94 for more manageable planning/tracking and sizing.

### PR DESCRIPTION
**Pull request by anngvu**
- "Master" issue: https://github.com/nf-osi/nfportalutils/issues/94
  - Covers about 20% of the refactor
  - Update tests and access-utils related vignette
- Also add vignette for #172
  - 30% of the work outlined in this issue

Update:
- Currently, have to be very specific about using `synapser` (>= 1.0.0) and < 2.0, suggesting you use `1.3.0`, see linked issue discussion regarding `synSetAnnotations`. However, by the time everything is refactored, I trust that `synapser` will have added back `synSetAnnotations`.
- Ignore the one failed MacOS check, I think it is a `brew` issue upstream that will be resolved eventually, see discussion at https://github.com/r-lib/actions/issues/280
image
- Suggest @allaway can review the low-level code, @cconrad8 can be made aware of the new vignettes.

### CODE REVIEW

**Comment by anngvu** [comment]
There appears to be a `synapser` issue that has to be resolved upstream of us. So on Linux stable source build, can use `synapser::synSetAnnotations` just fine. On MacOs and Windows, one gets the error 
`Error: 'synSetAnnotations' is not an exported object from 'namespace:synapser'`

**Comment by allaway** [comment]
@anngvu, should we report this on Sage-Bionetworks/synapser?

**Comment by anngvu** [comment]
Reported in https://github.com/Sage-Bionetworks/synapser/issues/331. I think for now best solution is to specify `synapser` (>= 1.0.0) and < 2.0 and update CI workflow.

**Comment by anngvu** [comment]
Update 2:
- #176, especially helpful for this PR and upcoming tutorials.
- Partial work on #177 as another vignette.

File: R/access_utils.R [hunk 1]
```diff
@@ -29,11 +29,11 @@ summarize_file_access <- function(principal_id, # 3378999 for NF-OSI
                                   fileview_id # "syn16858331"
                         ) {
 
-  .check_login()
   tryCatch({
-    view <- .syn$tableQuery(glue::glue("SELECT id,type,benefactorId FROM {fileview_id}"))
+    view <- synapser::synTableQuery(glue::glue("SELECT id,type,benefactorId FROM {fileview_id}")) %>%
+      synapser::as.data.frame() %>%
```
Comment (anngvu) [inline R/access_utils.R:34]: Actually do not need `synapser::`

**Comment by anngvu** [comment]
Merging this into [`develop-synapser`](https://github.com/nf-osi/nfportalutils/tree/develop-synapser) isntead of main `develop` for now.

**Pull request merged**